### PR TITLE
Public function to register custom ops

### LINKED ISSUE #1151: Public API to force load custom ops
Currently, it is inconvenient to load in Python a SavedModel that includes Addons custom ops. Consider the example below:

* `save.py`

```python
import tensorflow as tf
import tensorflow_addons as tfa

class Model(tf.keras.Model):
    @tf.function(input_signature=(tf.TensorSpec(shape=[None, 32], dtype=tf.float32),))
    def call(self, x):
        return tfa.activations.gelu(x)

model = Model()
tf.saved_model.save(model, '/tmp/model', signatures=model.call)
```

* `load.py`

```python
import tensorflow as tf
tf.saved_model.load("/tmp/model")
```

The load will fail because Addons custom ops are not registered to the TensorFlow runtime. This is expected as we first have to invoke `tf.load_op_library` on the custom ops.

However, with the new work on lazy loading #855 it got harder to force this op registration. For this model, the user should run the following which relies on internal APIs:

```python
from tensorflow_addons.activations.gelu import _activation_so
_activation_so.ops
```

If the custom ops are not loaded during the main import (i.e. during `import tensorflow_addons`), then the package should expose a public API that registers all custom ops.

Any thoughts?

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Fix #1151

I'll open an issue to track the registering of keras objects with this function. It's there for future backward compatibility.

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Haven't had time for a full review, but at first glance looks like a good strategy. @guillaumekln could you take a look when time allows and see if this covers the issue from your perspective.

File: tensorflow_addons/register.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import glob
+import os
+from pathlib import Path
+
+import tensorflow as tf
+
+from tensorflow_addons.utils.resource_loader import get_project_root
+
+
+def register_all(keras_objects: bool = True, custom_kernels: bool = True) -> None:
```
Comment (guillaumekln) [inline tensorflow_addons/register.py]: Maybe you should add docstring here to describe what it does and when it is required?

File: tensorflow_addons/register.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            "use custom ops in your model, you can skip registering custom ops with "
+            "`tfa.register_all(custom_kernels=False)`".format(shared_object)
+        ) from e
+
+
+def _get_all_shared_objects():
+    custom_ops_dir = os.path.join(get_project_root(), "custom_ops")
+    all_shared_objects = glob.glob(custom_ops_dir + "/**/*.so", recursive=True)
+    all_shared_objects = [x for x in all_shared_objects if Path(x).is_file()]
+    return list(all_shared_objects)
```
Comment (guillaumekln) [inline tensorflow_addons/register.py]: Looks like you don't need to copy the list here.

**Comment by gabrieldemarmiesse** [comment]
Thanks a lot for the review! Let me fix all that.

**Comment by guillaumekln** [comment]
@seanpmorgan Do you want to take a second look before merging?

**Comment by gabrieldemarmiesse** [comment]
This pull request will be discussed at the meeting tonight I believe.

**Comment by seanpmorgan** [comment]
@gabrieldemarmiesse Conflicts and then LGTM. Per discussion we'll want to work toward registering Keras objects in a similar manner

**Comment by gabrieldemarmiesse** [comment]
> Per discussion we'll want to work toward registering Keras objects in a similar manner

Let's do that in another pull request to avoid making the review harder.

**Pull request merged**